### ISSUE TITLE
AssertionError

### ISSUE THREAD

**Issue by lsp-stack** [Aldrich2y/MIANet#2]
Hello author, I have encountered some problems.
This is my pascal_split0_resnet50.yaml :
`DATA:
  classes: 2
  data_set: pascal  # coco

TRAIN:
  layers: 50
  sync_bn: False
  train_h: 473
  train_w: 473
  val_size: 473
  scale_min: 0.9  # minimum random scale
  scale_max: 1.1 # maximum random scale
  rotate_min: -10  # minimum random rotate
  rotate_max: 10  # maximum random rotate
  zoom_factor: 8  # zoom factor for final prediction during training, be in [1, 2, 4, 8]
  ignore_label: 255
  padding_label: 255
  aux_weight: 1.0
  train_gpu: [0]
  workers: 8  # data loader workers
  batch_size: 4  # batch size for training
  batch_size_val: 1
  base_lr: 0.005
  epochs: 200
  start_epoch: 0
  power: 0.9 # 0 means no decay
  momentum: 0.9
  weight_decay: 0.0001
  manual_seed: 1234    #1234
  print_freq: 5
  save_freq: 20
  save_path: exp/pascal/split0_resnet50/model
  weight:
  resume:
  evaluate: True
  split: 0
  shot: 1
  vgg: True
  ppm_scales: [60, 30, 15, 8]
  fix_random_seed_val: True
  warmup: False
  use_split_coco: True
  resized_val: True
  ori_resize: True  # use original label for evaluation

## deprecated multi-processing training
Distributed:
  dist_url: tcp://127.0.0.1:6789
  dist_backend: 'nccl'
  multiprocessing_distributed: False
  world_size: 1
  rank: 0
  use_apex: False
  opt_level: 'O0'
  keep_batchnorm_fp32:
  loss_scale:`
After I ran train.py ,something was wong:
![image](https://github.com/Aldrich2y/MIANet/assets/85109177/9cd8e767-4575-4fad-a7f1-2e6d79714942)
Following the settings above, I found that in the function __getitem__ of the dataset.py,
` for c in label_class:
            if c in self.sub_val_list:
                if self.mode == 'val' or self.mode == 'demo' or self.mode == 'finetune':
                    new_label_class.append(c)
            if c in self.sub_list:
                if self.mode == 'train':
                    new_label_class.append(c)
        label_class = new_label_class
        assert len(label_class) >0`
The values of list label_class were [0,37,147,150] ,the values of list sub_val_list were [1,2,3,4,5] and the values of list sub_list were [6,7,8,9,10,11,12,13,14,15,16,17,18,19,20].After running the above code, label_class's length becomes zero.I think there might be a bug here.Can you tell me why this problem occurs and how to fix it?

**Comment by Aldrich2y** [comment]
Hello. It seems your GT dataset is not numbered according to labels 1-20, I suggest you double check the files: MIANet/data/OCdevkit2012/SegmentationClassAug  and MIANet/util/dataset.

**Comment by lsp-stack** [comment]
Thank you for your help,I reprocessed my dataset. But it can only run successfully before the start validation of the first epoch.Do you think there are still mistakes in my dataset?

**Comment by Aldrich2y** [comment]
Hello, please describe in more detail.
